### ISSUE TITLE
Does it support "paging" feature?

### ISSUE THREAD

**Issue by ElliotMok** [mybatis/mybatis-dynamic-sql#29]
I didn't saw any methods like 'limit' or 'offset' in `org.mybatis.dynamic.sql.sqlBuilder`... Does it support "paging" feature natively or can be extend to support?

**Comment by jeffgbutler** [comment]
There's nothing built-in that does paging now.  I'm trying to avoid the dilemma of differing SQL dialects and just sticking to "standard" SQL for now.

I did write about how to implement a paging query at the bottom of this page: http://www.mybatis.org/mybatis-dynamic-sql/docs/whereClauses.html

**Comment by ElliotMok** [comment]
@jeffgbutler  By the way, it's said that "This library grew out of a desire to create a utility that could be used to improve the code generated by MyBatis Generator".So the question below is more related to MBG.
I found that not just XXXXStatementProvider classes but more XXXDSL classes were be used among the underlying code of the newest version of MBG.But the official website didn't cover too much about class like XXXDSL,FromGatherer,MyBatis3SelectModelAdapter etc.So is it possible to generate mapper method with paging feature like:
```
List<Lecture> records = tLecturesMapper.selectByExampleWithLimitAndOffset(2, 3)
                .where(title, isLike("%어린이%"), or(title, isLike("%습진")))
                .and(ispublish, isEqualTo(false))
                .orderBy()
                .build().execute();
```
or

```
List<Lecture> records = tLecturesMapper.selectByExample()
                .where(title, isLike("%어린이%"), or(title, isLike("%습진")))
                .and(ispublish, isEqualTo(false))
                .orderBy()
                .limit(5)          //this way looks more compatible with other mapper methods?
                .offset(1)
                .build().execute();
```

Any suggestion?

**Comment by jeffgbutler** [comment]
Here's a gist with an idea of how to do this without a too much difficulty: https://gist.github.com/jeffgbutler/5df477b4f72f5461a8cc32fb7cf4669b

I'll think about some other options too.

**Comment by jeffgbutler** [comment]
I made some changes to allow you to create a mapper method like your first example.  You can see how to do that here: https://github.com/mybatis/mybatis-dynamic-sql/tree/master/src/test/java/examples/paging

Currently there is no extension point to write queries like the second method.

You'll be able to do this once 1.0.1 is released - should be soon.

**Comment by jeffgbutler** [comment]
Resolved via #30
